### PR TITLE
Upgrade typescript

### LINKED ISSUE #20: Upgrade typescript version
Also upgrade `react-intl` as it requires higher version of typescript that we currently have.

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR upgrades typescript & react-intl (and corresponding libraries)

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

- update tsconfig file because of this:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-use-before-define.md#how-to-use

**Deletions** ⚰️

- delete `@formatjs/intl-unified-numberformat` as we are not using it anymore (it was deprecated and a different package created instead but we don't need it)

resolves #20

### CODE REVIEW

File: src/redux/immutable/FetchableData.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ const defaultValues: DefaultValues<Shared> = {
 };
 
 export class FetchableDataRecord<R> extends Record<Shared>(defaultValues) {
-  get record() {
+  get getRecord() {
```
Comment (rdig) [inline src/redux/immutable/FetchableData.ts:22]: Why was this renamed ?

File: src/redux/immutable/FetchableData.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ const defaultValues: DefaultValues<Shared> = {
 };
 
 export class FetchableDataRecord<R> extends Record<Shared>(defaultValues) {
```
Comment (rdig) [inline src/redux/immutable/FetchableData.ts:21]: I think there's a good case to be made to get rid of the whole fetchable data concept and it's not really relevant to us anymore

**Review by rdig — APPROVED** [review]
Looks good so far, but we might need to come back to it once the CDapp is up and running

File: src/redux/immutable/FetchableData.ts (hunk 1 above)
Comment (chinins) [reply src/redux/immutable/FetchableData.ts:22]: Because it extends the interface below
```interface Shared {
  record: any;
  error?: string | void;
  isFetching?: boolean;
  lastFetchedAt?: Date;
}```
And it gives the following error
`(getter) FetchableDataRecord<R>.record: R
'record' is defined as a property in class 'Record<Shared> & Readonly<Shared>', but is overridden here in 'FetchableDataRecord<R>' as an accessor.ts(2611)`

File: src/redux/immutable/FetchableData.ts (hunk 2 above)
Comment (chinins) [reply src/redux/immutable/FetchableData.ts:21]: Should I remove it in this PR or should I create a separate issue to do that?

Comment (rdig) [reply src/redux/immutable/FetchableData.ts:21]: I need to test it properly / think about it harder :)

So open a separate issue for it :)
